apps/src/tutorialExplorer/tutorialExplorer.js

```diff
@@ -0,0 +1,374 @@
+      // No filters set for a category, then show everything that might match.
+      // Tutorial has no tags, then it'll show.
+      // But if we actually have filters for a category, and the tutorial does too,
+      // then at least one filter must have a tag.
+      //   e.g. if the user chooses two platforms, then at least one of the
+      //   platforms must match the tutorial.
+
+      function filterFn(tutorial, index, array) {
+
+        // First check that the tutorial language doesn't exclude it immediately.
+        // If the tags contain some languages, and we don't have a match, then
+        // hide the tutorial.
+        if (tutorial.languages_supported) {
+          var languageTags = tutorial.languages_supported.split(',');
+          var currentLocale = this.props.locale;
+          if (languageTags.length > 0 &&
+            !languageTags.includes(currentLocale) &&
+            !languageTags.includes(currentLocale.substring(0,2))) {
+            return false;
+          }
+        }
+
+        // If we miss any filter group, then we don't show the tutorial.
+        var filterGroupMiss = false;
+
+        for (var filterGroupName in this.props.filters) {
+          var tutorialTags = tutorial["tags_" + filterGroupName];
+          if (tutorialTags && tutorialTags.length > 0) {
+            var tutorialTagsSplit = tutorialTags.split(',');
+
+            // now check all the filter group's tags
+            var filterGroup = this.props.filters[filterGroupName];
+
+            // For this filter group, we've not yet found a matching tag between
+            // user selected otions and tutorial tags.
+            var filterHit = false;
+
+            for (var filterName of filterGroup) {
+              if (tutorialTagsSplit.includes(filterName)) {
+                // The tutorial had a matching tag.
+                filterHit = true;
+              }
+            }
+
+            // The filter group needs at least one user-selected filter to hit
+            // on the tutorial.
+            if (filterGroup.length !== 0 && !filterHit) {
+              filterGroupMiss = true;
+            }
+          }
+        }
+
+        return !filterGroupMiss;
+      }
+
+      return (
+        <div className="col-80" style={{float: 'left'}}>
+          {this.props.tutorials.filter(filterFn, this).map(item => <Tutorial item={item} filters={this.props.filters} key={item.code}/>)}
+        </div>
+      );
+    }
+  });
+
+  const TutorialExplorer = React.createClass({
+    propTypes: {
+      filterGroups: React.PropTypes.array,
+      tutorials: React.PropTypes.array,
+      locale: React.PropTypes.string
+    },
+
+    getInitialState: function () {
+      var filters = {};
+
+      for (let filterGroup of this.props.filterGroups) {
+        filters[filterGroup.name] = [];
+
+      }
+      return {
+        filters: filters
+      };
+    },
+
+    handleUserInput: function (filterGroup, filterEntry, value) {
```

Comment:
Could use a function description, including perhaps @param tags describing the inputs.